### ISSUE TITLE
cookbook_version is undefined from a Chef 11 client

### ISSUE THREAD

**Issue by metadave** [chef/chef-analytics#6]
```
error: ECMA 262 regex "^[0-9]*\.[0-9]*(\.[0-9]*)?$" does not match input string "undefined"
    level: "error"
    schema: {"loadingURI":"#","pointer":"/properties/resources/items/properties/cookbook_version"}
    instance: {"pointer":"/resources/1/cookbook_version"}
    domain: "validation"
    keyword: "pattern"
    regex: "^[0-9]*\\.[0-9]*(\\.[0-9]*)?$"
    string: "undefined"
```

**Comment by metadave** [comment]
@sersut is this something you can help with?

**Comment by sersut** [comment]
Sure... Is this a reporting issue? What is the steps to repro this?

**Comment by metadave** [comment]
it's from an ec-metal run based on this branch: https://github.com/chef/ec-metal/tree/irving/loadtesting_analytics, not sure if I have any more info than that :-(

**Comment by jamesc** [comment]
So this is where we fixed this in Chef client -  https://github.com/chef/chef/pull/888.  Before this we would get null cookbook_version/cookbook_name if the resource had been created dynamically.

**Comment by sersut** [comment]
Ah nice. Looks like we don't have an action item on this. The fix would be to upgrade to 11.18.0. 

Not sure how ec-metal sets this. Unless this is a repro with 11.18.0 we can close this I believe.

**Comment by jamesc** [comment]
Closing. Fix as aboce
